pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -801,7 +802,7 @@ func copyFiles(runner command.Runner, files []assets.CopyableFile) error {
 	return nil
 }
 
-func startKubeletIfRequired(runner command.Runner, sm sysinit.Manager) error {
+func startKubelet(runner command.Runner, sm sysinit.Manager) error {
```

Comment:
I think this function can be removed and replaced entirely with calls to `sm.Enable("kubelet")` now.

Reply:
I would prefer to keep it as a separate function, because it's called in `UpdateNode` which is a bit long & I think separating it into clear steps is more readable

Reply:
`sm.Start("kubelet")` is as many lines as `startKubelet`

If you feel strongly, please just replace this function with a call to `sm.Start("kubelet")`. The rest of what it does is no longer meaningful.

Reply:
Ah, does that mean we no longer need to copy over the kubelet systemd files over in this function? I thought that was the purpose of this function, but if they're copied over elsewhere then I'm all for removing this

Reply:
Found where they copy over & removed the function, thank you!

Reply:
You'll need to fix `UpdateNode` so that it installs the files into `KubeletServiceFile` rather than `KubeletServiceFile+".new"`.

The reason for the complexity is that previously we were only restarting the kubelet if the configuration has changed on disk. Now we're completely controlling the kubelet lifecycle, so we will always start the kubelet when the node is coming up.